src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java

```diff
@@ -831,6 +877,24 @@ private void updateCaseBuilderForTransformCase(CaseData caseData, ResponseCaseDa
                     .willNumberOfCodicils(null);
         }
 
+        if (isApplyAsExecutor(caseData)) {
+            if (YES.equals(caseData.getSolsSolicitorIsMainApplicant())) {
```

Comment:
see if you can refactor this function into its own method for both calls

Reply:
Ends up making code less readable; having to pass multiple mutable objects as arguments